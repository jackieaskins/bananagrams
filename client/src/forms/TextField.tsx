import { FormControl, FormLabel, Input } from "@chakra-ui/react";

export interface TextFieldProps {
  isRequired?: boolean;
  label?: React.ReactNode;
  name: string;
  setValue: (newValue: string) => void;
  type?: React.HTMLInputTypeAttribute;
  value: string;
}

export default function TextField({
  isRequired = false,
  label,
  name,
  setValue,
  type,
  value,
}: TextFieldProps): JSX.Element {
  return (
    <FormControl isRequired={isRequired}>
      {label && <FormLabel htmlFor={name}>{label}</FormLabel>}
      <Input
        id={name}
        type={type}
        value={value}
        onChange={(event) => setValue(event.target.value)}
        name={name}
      />
    </FormControl>
  );
}
