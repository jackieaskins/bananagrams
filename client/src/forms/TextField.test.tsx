import { Input } from "@chakra-ui/react";
import { shallow } from "enzyme";
import TextField from "./TextField";

describe("<TextField />", () => {
  const mockSetValue = jest.fn();

  const renderComponent = () =>
    shallow(
      <TextField
        setValue={mockSetValue}
        value="value"
        isRequired
        label="Label"
        name="name"
        type="text"
      />,
    );

  test("renders properly", () => {
    expect(renderComponent()).toMatchSnapshot();
  });

  test("onChange calls setValue", () => {
    renderComponent()
      .find(Input)
      .props()
      .onChange({ target: { value: "newValue" } });

    expect(mockSetValue).toHaveBeenCalledWith("newValue");
  });
});
