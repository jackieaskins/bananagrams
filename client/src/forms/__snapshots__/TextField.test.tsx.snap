// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders properly 1`] = `
<FormControl
  isRequired={true}
>
  <FormLabel
    htmlFor="name"
  >
    Label
  </FormLabel>
  <Input
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
    value="value"
  />
</FormControl>
`;
